ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/RDXRawImagePointCloudRenderer.java

```diff
@@ -0,0 +1,227 @@
+import org.bytedeco.opencv.opencv_core.Mat;
+import org.jetbrains.annotations.Nullable;
+import us.ihmc.euclid.referenceFrame.FramePose3D;
+import us.ihmc.euclid.referenceFrame.interfaces.FixedFramePose3DBasics;
+import us.ihmc.euclid.transform.RigidBodyTransform;
+import us.ihmc.perception.RawImage;
+import us.ihmc.perception.camera.CameraIntrinsics;
+import us.ihmc.rdx.AbstractRDXPointCloudRenderer;
+import us.ihmc.rdx.shader.RDXShader;
+import us.ihmc.rdx.shader.RDXUniform;
+import us.ihmc.rdx.tools.LibGDXTools;
+
+import java.nio.FloatBuffer;
+
+import static us.ihmc.rdx.AbstractRDXPointCloudRenderer.ColoringMethod.*;
+
+public class RDXRawImagePointCloudRenderer extends AbstractRDXPointCloudRenderer
+{
+   private final VertexAttributes vertexAttributes = new VertexAttributes(new VertexAttribute(Usage.Generic, 1, "a_depthData"));
+
+   // DEPTH IMAGE UNIFORM DATA
+   private CameraIntrinsics depthIntrinsics = new CameraIntrinsics();
+   private float depthDiscretization = 0.0f;
+   private final FixedFramePose3DBasics depthPose = new FramePose3D();
+   private final Matrix4 libGDXDepthTransform = new Matrix4();
+   private final RigidBodyTransform tempDepthTransform = new RigidBodyTransform();
+
+   // COLOR IMAGE UNIFORM DATA
+   private Pixmap colorImagePixmap;
+   private BytePointer pixmapDataPointer;
+   private Texture colorImageTexture;
+   private CameraIntrinsics colorIntrinsics = new CameraIntrinsics();
+   private final RigidBodyTransform depthToColorTransform = new RigidBodyTransform();
+   private final Matrix4 libGDXColorTransform = new Matrix4();
+   private final RigidBodyTransform tempColorTransform = new RigidBodyTransform();
+
+   private final boolean enableColorImageInput;
+
+   public RDXRawImagePointCloudRenderer(boolean enableColorImageInput)
+   {
+      this.enableColorImageInput = enableColorImageInput;
+   }
+
+   public void updateMesh(RawImage depthImage)
+   {
+      if (depthImage.get() == null)
+         return;
+
+      // Update depth image uniforms
+      depthIntrinsics = depthImage.getIntrinsicsCopy();
+      depthDiscretization = depthImage.getDepthDiscretization();
+      depthPose.set(depthImage.getPose());
+      LibGDXTools.toLibGDX(depthPose, tempDepthTransform, libGDXDepthTransform);
+
+      int width = depthImage.getWidth();
+      int height = depthImage.getHeight();
+      int pixelCount = width * height;
+
+      FloatBuffer verticesBuffer = renderable.meshPart.mesh.getVerticesBuffer(true); // Mark dirty
+
+      // Ensure correct length and initialize vertices buffer
+      if (renderable.meshPart.size != pixelCount)
+      {
+         renderable.meshPart.size = pixelCount;
+         verticesBuffer.limit(pixelCount);
+      }
+
+      // Wrap the vertices buffer into a pointer, than into a Mat
+      FloatPointer verticesPointer = new FloatPointer(verticesBuffer);
+      Mat verticesMat = new Mat(height, width, opencv_core.CV_32FC1, verticesPointer);
```

Comment:
maybe add a comment about why we do this

Reply:
Added a comment. Let me know if I should edit it.